### PR TITLE
always refresh ocShares prior listing them to have an up-to-date ocFile

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
The expected result of this PR is to have a sharedView that does not crash any longer.
(clicking on an image might still crash or result in a wrong image due to #857)

Ref: #869

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/adapter/FileListListAdapter.java [hunk 1]
```diff
@@ -491,9 +493,19 @@ public void setData(ArrayList<Object> objects, ExtendedListFragment.SearchType s
 
                     shares.add(ocShare);
 
-                    OCFile ocFile = mStorageManager.getFileByPath(ocShare.getPath());
-                    if (!mFiles.contains(ocFile)) {
-                        mFiles.add(ocFile);
+                    // get ocFile from Server to have an up-to-date copy
+                    ReadRemoteFileOperation operation = new ReadRemoteFileOperation(ocShare.getPath());
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/adapter/FileListListAdapter.java]: Please use search api instead to get all the properties for all the relevant files in 1 network call. You might need to extend the search magic in the library for this though, but that would be significantly better.

Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/adapter/FileListListAdapter.java]: As of feature freeze on server the enhancement will not be available for NC12, so I opened up #879 and this shall be merged, though it is not perfect :/

**Comment by mario** [comment]
👍 

@AndyScherzinger please review and merge.

**Comment by AndyScherzinger** [comment]
👍 Looks good! :)

**Pull request merged**